osgp/platform/osgp-adapter-domain-smartmetering/src/test/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/services/InstallationServiceTest.java

```diff
@@ -121,18 +125,69 @@ public void addMeterProtocolInfoNotFound() {
   }
 
   @Test
-  public void addMeterDeviceExists() {
-
-    // SETUP
+  void addMeterDeviceExists() {
     when(this.smartMeteringDeviceRepository.findByDeviceIdentification(DEVICE_IDENTIFICATION))
         .thenReturn(this.smartMeter);
 
-    // CALL
     try {
       this.instance.addMeter(this.deviceMessageMetadata, this.addSmartMeterRequest);
     } catch (final FunctionalException e) {
       assertThat(e.getExceptionType()).isEqualTo(FunctionalExceptionType.EXISTING_DEVICE);
       assertThat(e.getComponentType()).isEqualTo(ComponentType.DOMAIN_SMART_METERING);
     }
   }
+
+  @Test
+  void testUpdateSmartMeter() throws FunctionalException {
+    final SmartMeter existingSmartMeter = new SmartMeter();
+    existingSmartMeter.setId(1L);
+    existingSmartMeter.setVersion(2L);
+
+    when(this.smartMeteringDeviceRepository.findByDeviceIdentification(DEVICE_IDENTIFICATION))
+        .thenReturn(existingSmartMeter);
+    when(this.addSmartMeterRequest.getOverwrite()).thenReturn(true);
+
+    doReturn(existingSmartMeter)
+        .when(this.smartMeterService)
+        .convertSmartMeter(this.smartMeteringDevice);
+
+    when(this.protocolInfoRepository.findByProtocolAndProtocolVersionAndProtocolVariant(
+            this.smartMeteringDevice.getProtocolName(),
+            this.smartMeteringDevice.getProtocolVersion(),
+            this.smartMeteringDevice.getProtocolVariant()))
+        .thenReturn(this.protocolInfo);
```

Comment:
These lines of codes are equals with the next test except for the overwrite boolean. Did you consider a parameterized test?

Reply:
I use the ParameterizedTest with overwrite boolean in the smart-meter-headend pr, but in some unexplainable way not in this one. They are all updated and merged now.